src/Query.tsx

```diff
@@ -0,0 +1,209 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import ApolloClient, {
+  ObservableQuery,
+  ApolloQueryResult,
+  ApolloError,
+  FetchMoreOptions,
+  UpdateQueryOptions,
+  FetchMoreQueryOptions,
+} from 'apollo-client';
+import { DocumentNode } from 'graphql';
+import { ZenObservable } from 'zen-observable-ts';
+
+import shallowEqual from './shallowEqual';
+const invariant = require('invariant');
+const pick = require('lodash.pick');
+
+import { QueryOpts, OperationVariables } from './types';
+import { parser, DocumentType } from './parser';
+
+export interface QueryRenderProp {
```

Comment:
I think we should rename this type to prevent confusion. I suggest:
- `QueryResult` (preferred)
- `QueryRenderArgs`